modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js

```diff
@@ -213,18 +221,20 @@ const DatePicker = ({
 				showMask: true,
 			});
 
-			if (localizedValue[locale]) {
+			const currentValue = localizedValue[locale];
+
+			if (currentValue) {
 				if (
-					typeof localizedValue[locale] === 'string' &&
-					(localizedValue[locale].includes('/') ||
-						localizedValue[locale].includes('.'))
+					typeof currentValue === 'string' &&
+					currentValue.includes('_') &&
+					currentValue.match(DIGIT_REGEX)
 				) {
-					inputRef.current.value = localizedValue[locale];
+					inputRef.current.value = currentValue;
 				}
 				else {
-					inputRef.current.value = moment(
-						localizedValue[locale]
-					).format(dateMask.toUpperCase());
+					inputRef.current.value = moment(currentValue).format(
+						dateMask.toUpperCase()
+					);
 				}
 			}
```

Comment:
A suggestion you to simplify the code:
```javascript
const currentValue = localizedValue[locale];

if (currentValue) {
	inputRef.current.value = typeof currentValue === 'string' && currentValue.includes('_') && currentValue.match(DIGIT_REGEX)
		? currentValue
		: moment(currentValue).format(dateMask.toUpperCase());	
}
```

But i think you can also simplify the second `if` to verify if the string has the underscore (`_`) only:
 ```javascript
const currentValue = localizedValue[locale];

if (currentValue) {
	inputRef.current.value = /_/.test(currentValue)
		? currentValue
		: moment(currentValue).format(dateMask.toUpperCase());	
}
```

Reply:
Thanks for the simplification, @miltonmc!
I'll just keep the second condition because it fixes the cursor position when the user cleans the field. `inputRef.current.value = /_/.test(currentValue) && currentValue.match(DIGIT_REGEX)`